cnf-certification-test/operator/suite.go

```diff
@@ -303,3 +312,54 @@ func testOperatorSingleCrdOwner(check *checksdb.Check, env *provider.TestEnviron
 
 	check.SetResult(compliantObjects, nonCompliantObjects)
 }
+
+func testOperatorSecurityRequiremnents(check *checksdb.Check, env *provider.TestEnvironment) {
```

Comment:
I guess this is more of an organization question, but what do we think about splitting this into 4 separate tests?  Since we are setting compliant/non-compliant objects depending on multiple factors, wouldn't it be more organizational to have completely separate tests for these different test cases?

Reply:
For example, instead of `testOperatorSecurityRequiremnents()`, we would have:
- testOperatorRunAsUserID()
- testOperatorRunAsNonRoot()
- testOperatorAutomountTokens()
- testOperatorReadOnlyFilesystem()

Reply:
I think this is a good idea and it can help in the future. I will split it into 4 functions. @sebrandon1

Reply:
@sebrandon1 I was thinking to have a list of security requirements for any pod (the listed ones in the JIRA) named as getAllPodsWithSecurityRequirements(), instead of having a new test case for each of the requirement. Apply those requirements to any pods (operator, specific application, namespace pods) and then get the test case passed and say that these pods are satisfying the requirements. If we list 5 items (which can change later also if we want to add more restrictive requirements) as security requirements, then any pod that passed all of them are called to satisfy the requirement. Otherwise not. IMO, the requirements for security comes as a block, not separately.